frontend/document/document.lua

```diff
@@ -207,8 +207,51 @@ function Document:getNativePageDimensions(pageno)
     return page_size
 end
 
-function Document:getProps()
-    return self._document:getDocumentProps()
+function Document:getDocumentProps()
+    -- pdfdocument, djvudocument
+    return self._document:getMetadata()
+    -- credocument, picdocument - overridden by a document implementation
+end
+
+function Document:getProps(props)
+    local function makeNilIfEmpty(str)
+        if str == "" then
+            return nil
+        end
```

Comment:
It is much more effective in the code to have `nil` instead of `""` for empty props.
For the purpose of (future) custom metadata, one or more whitespaces are not an empty prop.